### PR TITLE
feat: add color utility package

### LINKED ISSUE #137: Add color utility package
Should contain "range"/"variant" function for creating ranges of colors from a single color. Work with Bobby and Martin to get details on the algorithm.

### PR DESCRIPTION
**Pull request by nicholasrice**
closes #137

### CODE REVIEW

File: packages/fast-colors/README.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+```js
+import { range } from "@microsoft/fast-colors";
+
+// Default usage
+const range = range("#dddddd"); // [ '#f2f2f2', '#ebebeb', '#e4e4e4', '#dddddd', '#b6b6b6', '#8f8f8f', '#696969' ]
+
+// Custom configuration
+const range = range("#dddddd", {
+    count: 3,
+    padingLight: .2,
```
Comment (chrisdholt) [inline packages/fast-colors/README.md]: nit: spelling

"padding"

File: packages/fast-colors/package.json [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  "description": "A color utility for generating a range of colors from a single color",
+  "license": "MIT",
+  "main": "./dist/index.js",
+  "types": "./dist/index.d.ts",
+  "contributors": [
+    {
+      "name": "Fluent Web",
+      "email": "fwq@microsoft.com"
+    },
+    {
```
Comment (chrisdholt) [inline packages/fast-colors/package.json]: Are we doing this? May be difficult to maintain as more contributors jump in. I thought we were going to do inline and in readme? Just curious.

Comment (nicholasrice) [reply packages/fast-colors/package.json]: Yea not sure - I thought I'd call them out directly as this is essentially a port of their codebase but we can discuss. Thoughts Aaron?

Comment (chrisdholt) [reply packages/fast-colors/package.json]: @awentzel ^ thoughts?

Comment (awentzel) [reply packages/fast-colors/package.json]: We were going add a section in the readme to call them out specifically.  The package.json file then doesn't have to change. Additionally, I will add them as contributors to the project.

Comment (awentzel) [reply packages/fast-colors/package.json]: You probably will want to ask them what email/profile to use for their work.

Comment (nicholasrice) [reply packages/fast-colors/package.json]: Okay sounds good - I'll add it to the Readme. I discussed with bobby point to their GitHub handles, so I'll reference their profile pages

File: packages/fast-colors/src/range.spec.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+const white: string = "#ffffff";
+
+describe("range", (): void => {
+    test("should return 7 range by default", (): void => {
+        expect(range(white).length).toBe(7);
+    });
+    test("should return a custom number of range", (): void => {
+        expect(expect(range(white, { count: 2 }).length).toBe(2));
+    });
+    test("should create range from a hexidecimal color", (): void => {
```
Comment (mlijanto) [inline packages/fast-colors/src/range.spec.ts]: nit: typo `hexadecimal` :)

File: packages/fast-colors/README.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+## range(color, [options])
+A function that converts a single color input into a range of outputs, from lightest to darkest. 
+- `color: string` - a HEX or RGB color represented as a string (eg, `"#FFF"` or `"rgb(0, 0, 0)"`)
+- `options: object`
+    - `count: number` - the number of colors to create in the count. Defaults to `7`.
+    - `paddingLight: number` - the amount to reduce the color-range of light values by. Defaults to `0.185`.
+    - `paddingDark: number` - the amount to reduce the color-range of dark values by. Defaults to `0.185`.
+    - `saturationLight: number` - the amount to adjust the saturation of light values by. Defaults to `0.35`.
+    - `saturationDark: number` - the amount to adjust the saturation of light values by. Defaults to `1.25`.
```
Comment (mlijanto) [inline packages/fast-colors/README.md]: Looks like the `light` in this sentence should be `dark` instead.

File: packages/fast-colors/package.json (hunk 2 above)
Comment (nicholasrice) [reply packages/fast-colors/package.json]: @chrisdholt @awentzel Updates made.

File: packages/fast-colors/src/range.spec.ts (hunk 3 above)
Comment (nicholasrice) [reply packages/fast-colors/src/range.spec.ts]: Good eye, thanks Martin!

File: packages/fast-colors/README.md (hunk 4 above)
Comment (nicholasrice) [reply packages/fast-colors/README.md]: Fixed

**Pull request merged**